File: internal/model/track.go
package model

import "time"

// Track represents a single audio track in the library.
type Track struct {
	ID              string    `json:"id"`
	AlbumID         string    `json:"albumId"`
	ArtistID        string    `json:"artistId"`
	Title           string    `json:"title"`
	TrackNumber     int       `json:"trackNumber,omitempty"`
	DiscNumber      int       `json:"discNumber"`
	DurationSeconds int       `json:"durationSeconds"`
	FilePath        string    `json:"-"`
	FileSize        int64     `json:"fileSize,omitempty"`
	Format          string    `json:"format,omitempty"`
	Bitrate         int       `json:"bitrate,omitempty"`
	SampleRate      int       `json:"sampleRate,omitempty"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

File: ios/Sonus/Models/Track.swift
import Foundation

/// A single audio track in the library.
struct Track: Codable, Identifiable, Hashable, Sendable {
    let id: String
    let albumId: String
    let artistId: String
    let title: String
    let trackNumber: Int?
    let discNumber: Int
    let durationSeconds: Int
    let fileSize: Int64?
    let format: String?
    let bitrate: Int?
    let sampleRate: Int?
    let createdAt: Date
    let updatedAt: Date
}
